ApiClient.js

```diff
@@ -20,7 +40,10 @@ export default class ApiClient {
 			});
 	}
 
-	static post(endpoint, headers, body) {
+	static async post(endpoint, headers, authorized, body) {
```

Comment:
Something about having to pass a boolean for a specific task like auth as a standalone argument makes me hurt a little inside, especially for instances where we need to pass an empty headers object, ie `api.post('/some/endpoint', {}, true, body_data)`.

I definitely do like the architecture of each request specifying whether it should be authenticated and how `ApiClient` handles tokens centrally (nice work Stephen!), just wonder if we might want to consider switching up the signature for our Api methods to have an `options` parameter to hold data like headers and booleans. i.e:
```javascript
static async post(endpoint, body, options={}) {
  if (options.authenticate) //...
}
```
invoked as `api.post('/some/endpoint', body_data, { authenticate: true })`, with a `headers` property in the third argument if needed (which looks a little messy but is rarely needed). This catch-all `options` object would give us a better set-up for other spiciness we might want to add to requests in future, without having to extend the arguments list and dodge the ones we don't need to satisfy the method signature.

HOWEVER, as it is right now it's not that bad (only 4 args) and likely won't get that much bigger, plus having a headers argument makes a fair amount of sense for an Api method. What do you think @scholvat and @picklechips?
We'd have to go through most api methods and invocations to change this so I'd be happy to merge it as-is for now and make a small PR for it after.

Reply:
Sounds good, t'is probably better to get 'er all done now. And thanks :wink: